### PR TITLE
Wasm plugin: Upgrade to the latest proxy-wasm-cpp-host and remove reference to WAVM

### PR DESCRIPTION
**Pull request by shukitchan**
Please note all things under plugins/experimental/wasm/lib/ are coming from proxy-wasm external repos

### CODE REVIEW

**Comment by shukitchan** [comment]
[approve ci RAT]

**Comment by shukitchan** [comment]
[approve ci rat]

**Comment by shukitchan** [comment]
[approve ci rat]

**Comment by shukitchan** [comment]
[approve ci]

**Review by randall — APPROVED** [review]

File: plugins/experimental/wasm/lib/src/wasm.cc [hunk 1]
```diff
@@ -23,54 +23,75 @@
...
   }
-  return result;
 }
 
 } // namespace
 
 std::string makeVmKey(std::string_view vm_id, std::string_view vm_configuration,
                       std::string_view code) {
-  return BytesToHex(Sha256({vm_id, vm_configuration, code}));
+  return Sha256String({vm_id, "||", vm_configuration, "||", code});
```
Comment (serrislew) [inline plugins/experimental/wasm/lib/src/wasm.cc:94]: Are these "| |" intentional to add to the hash key? Since it wasn't originally included?

**Pull request merged**